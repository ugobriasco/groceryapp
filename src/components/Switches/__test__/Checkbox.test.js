import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import Icon from  'react-native-vector-icons/MaterialIcons';

import {Checkbox, styles} from '../';

describe('Checkbox', () => {

	it('renders correctly', () => {
	const tree = renderer.create(<Checkbox />).toJSON();
	expect(tree).toBeTruthy();

	});

	it ('styles an object', () => {
		expect(typeof styles).toBe('object');
	});

	describe('size stzling props', () => {
		it('should render an icon with the selected size', () => {
			const rendered = renderer.create(<Checkbox size='50'/>);
			expect(rendered).toMatchSnapshot();	
		});
		it('should render an unchecked box if isChecked=false', () => {
			const rendered = renderer.create(<Checkbox isChecked={false}/>);
			expect(rendered).toMatchSnapshot();	
		});
		it('should render an unchecked box if isChecked=true', () => {
			const rendered = renderer.create(<Checkbox isChecked={true}/>);
			expect(rendered).toMatchSnapshot();	
		});
	});

	describe('onPress prop', () => {
		it('should trigger a function if pressed ', () => {
			const rendered = renderer.create(<Checkbox onPress={()=>console.log('foo')}/>);
			expect(rendered).toMatchSnapshot();	
		});
	});
});



