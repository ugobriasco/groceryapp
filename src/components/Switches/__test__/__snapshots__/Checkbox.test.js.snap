// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox onPress prop should trigger a function if pressed  1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#777",
            "fontSize": 30,
          },
          Array [
            Object {
              "marginRight": 10,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Checkbox size stzling props should render an icon with the selected size 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      onPress={undefined}
      style={
        Array [
          Object {
            "color": "#777",
            "fontSize": 50,
          },
          Array [
            Object {
              "marginRight": 10,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Checkbox size stzling props should render an unchecked box if isChecked=false 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      onPress={undefined}
      style={
        Array [
          Object {
            "color": "#777",
            "fontSize": 30,
          },
          Array [
            Object {
              "marginRight": 10,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Checkbox size stzling props should render an unchecked box if isChecked=true 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      onPress={undefined}
      style={
        Array [
          Object {
            "color": "#777",
            "fontSize": 30,
          },
          Array [
            Object {
              "marginRight": 10,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
